src/gui/owncloudgui.cpp

```diff
@@ -174,22 +174,31 @@ void ownCloudGui::slotTrayClicked( QSystemTrayIcon::ActivationReason reason )
         last_click.start();
     }
 
-    // A click on the tray icon should only open the status window on Win and
-    // Linux, not on Mac. They want a menu entry.
-#if !defined Q_OS_MAC
+    // Left click
     if( reason == QSystemTrayIcon::Trigger ) {
-        // Start settings if config is existing.
-        slotOpenSettingsDialog();
-    }
+        if (OwncloudSetupWizard::bringWizardToFrontIfVisible()) {
+            // brought wizard to front
+        } else if (_shareDialogs.size() > 0) {
+            // Share dialog(s) be hidden by other apps, bring them back
+            Q_FOREACH(const QPointer<ShareDialog> &shareDialog, _shareDialogs) {
+                Q_ASSERT(shareDialog.data());
```

Comment:
Minor: I still think Q_FOREACH is awkward and, unrelatedly, wonder why _shareDialogs has QPointer<...> values, since deleted dialogs are removed from the map anyway...